### ISSUE TITLE
reformat warmup

### ISSUE THREAD

**Issue by binmakeswell** [hpcaitech/ColossalAI#31]
Currently, warmup step and epoch are mixed-used in lr_scheduler, which is a little confusing.
If add warmup_step in lr_scheduler_hook, which is more user-friendly for users needing steps in multiple epochs, maybe the name of 'by_epoch' needs to be more clear, resulting in changes of other modules and examples.

**Comment by binmakeswell** [comment]
Updated.
